tests/integration/cassandra_test.go

```diff
@@ -1,408 +1,295 @@
+	"github.com/testcontainers/testcontainers-go"
 	"path/filepath"
-	"regexp"
 	"strings"
 	"testing"
 	"time"
 )
 
 const (
 	envCassandraVersion = "3.11.0"
-	timeout             = "30000"
+	timeout             = "5000"
 )
 
 var (
 	iName = "cassandra"
 
-	defaultContainer = "integration_nri-cassandra_1"
-
 	iVersion = "1.1.0"
 
-	defaultBinPath  = "/nri-cassandra"
-	defaultHostname = "cassandra"
-
-	schemaFolder = fmt.Sprintf("json-schema-files-%s", envCassandraVersion)
+	integrationBinPath       = "/nri-cassandra"
+	integrationContainerName = "integration_nri-cassandra_1"
+	cassandraContainerName   = "integration_cassandra_1"
 
-	// cli flags
-	container = flag.String("container", defaultContainer, "container where the integration is installed")
-	binPath   = flag.String("bin", defaultBinPath, "Integration binary path")
-
-	hostname = flag.String("hostname", defaultHostname, "cassandra hostname")
+	schemaDir = fmt.Sprintf("json-schema-files-%s", envCassandraVersion)
 )
 
-// Returns the standard output, or fails testing if the command returned an error
-func runIntegration(t *testing.T, envVars ...string) (string, string, error) {
-	t.Helper()
-
-	command := make([]string, 0)
-	command = append(command, *binPath)
-
-	var found bool
-	for _, envVar := range envVars {
-		if strings.HasPrefix(envVar, "HOSTNAME") {
-			found = true
-			break
-		}
-	}
-
-	if !found && hostname != nil {
-		command = append(command, "--hostname", *hostname)
-	}
+type CassandraTestSuite struct {
+	suite.Suite
+	compose *testcontainers.LocalDockerCompose
+}
 
-	stdout, stderr, err := helpers.ExecInContainer(*container, command, envVars...)
+func TestCassandraTestSuite(t *testing.T) {
+	suite.Run(t, new(CassandraTestSuite))
+}
 
-	if stderr != "" {
-		log.Debug("Integration command Standard Error: ", stderr)
-	}
+func (s *CassandraTestSuite) SetupSuite() {
+	s.compose = testutils.ConfigureCassandraDockerCompose()
 
-	return stdout, stderr, err
-}
+	err := testutils.RunDockerCompose(s.compose)
+	require.NoError(s.T(), err)
 
-func TestMain(m *testing.M) {
-	fmt.Println("Wait for cassandra to initialize...")
+	// Could not rely on testcontainers wait strategies here, as the server might be up but not reporting all mbeans.
+	s.T().Log("Wait for cassandra to initialize...")
 	time.Sleep(30 * time.Second)
+}
 
-	flag.Parse()
+func (s *CassandraTestSuite) TearDownSuite() {
+	s.compose.Down()
+}
 
-	result := m.Run()
+func (s *CassandraTestSuite) TestCassandraIntegration_ValidArguments() {
+	t := s.T()
 
-	os.Exit(result)
-}
+	testName := testutils.GetTestName(t)
 
-func TestCassandraIntegration_ValidArguments(t *testing.T) {
-	testName := helpers.GetTestName(t)
+	ctx, cancelFn := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancelFn()
 
-	stdout, stderr, err := runIntegration(t,
-		"CONFIG_PATH=/etc/cassandra/cassandra.yaml",
-		fmt.Sprintf("TIMEOUT=%v", timeout),
-		fmt.Sprintf("NRIA_CACHE_PATH=/tmp/%v.json", testName),
-	)
+	env := map[string]string{
```

Comment:
same advice as before, would the tests be more readable having testsCases and an init?

```
	validCondid:=initTests()
	cases := map[string]struct{
		configModifier func(*agent.InjectorConfig),
		caseName string,
		errExpected
		stdOutExpected
	}
```

I have the feeling we are testing similar things, for example [here](https://github.com/newrelic/newrelic-infra-operator/blob/92bde3625f9775c4dc31acb2e257dc4f826b2d9a/internal/mutator/pod/agent/injector_test.go#L52) I feel like it is clearer what is changing in each test 😃